File: TungstenCore/Models/Assignment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TungstenCore.Models
{
    public class Assignment : Entity<string>
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required, ForeignKey("Segment")]
        public string SegmentId { get; set; }
        public virtual Segment Segment { get; set; }

        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime EndTime { get; set; }
    }
}

File: TungstenCore/ClientApp/app/classes/assignment.ts
import { IAssignment } from '../interfaces/assignment';
import { ISegment } from '../interfaces/segment';

export class Assignment implements IAssignment{
    Id: string;
    Name: string;
    Description: string;
    SegmentId: string;
    Segment: ISegment;

    constructor(name: string, description: string, segmentid: string) {
        this.Name = name;
        this.Description = description;
        this.SegmentId = segmentid;
    }
}